torchaudio_contrib/layers.py

```diff
@@ -212,43 +212,46 @@ def __repr__(self):
         return self.__class__.__name__ + param_str1 + param_str2 + param_str3
 
 
-class StretchSpecTime(_ModuleNoStateBuffers):
+class TimeStretch(_ModuleNoStateBuffers):
     """
     Stretch stft in time without modifying pitch for a given rate.
 
     Args:
 
         rate (float): rate to speed up or slow down by. Defaults to 1.
-        hop_len (int): Number audio of frames between STFT columns.
+        hop_length (int): Number audio of frames between STFT columns.
             Defaults to 512.
         num_bins (int, optional): number of filter banks from stft.
             Defaults to 1025.
     """
 
-    def __init__(self, rate=1., hop_len=512, num_bins=1025):
-        super(StretchSpecTime, self).__init__()
+    def __init__(self, rate=1., hop_length=512, num_bins=1025):
+        super(TimeStretch, self).__init__()
 
         self.rate = rate
-        phi_advance = torch.linspace(
-            0, math.pi * hop_len, num_bins)[..., None]
+        phase_advance = torch.linspace(
+            0, math.pi * hop_length, num_bins)[..., None]
 
-        self.register_buffer('phi_advance', phi_advance)
+        self.register_buffer('phase_advance', phase_advance)
 
     def forward(self, complex_specgrams, rate=None):
         """
 
         Args:
             complex_specgrams (Tensor): complex spectrogram
                 (batch, channel, freq, time, complex=2)
-            rate (float or None)
+            rate (float or None): if no rate is passed, use fixed value.
```

Comment:
i think this is a bit risky, that we have `self.rate` then also taking another variable. how about we starting without `rate` in `forward()` so that the use cases of dynamically changing rate vs not would be exclusively by `Layers` and `Functional` respectively? Otherwise - if there's a strong reason to enable it here - i'd say the name should be something like `overriding_rate` to make it super clear that it's gonna override the `self.rate`.

Reply:
Hmm I'm not too sure I follow how to change the rate if we're not passing it to forward. I do like `overriding_rate` though.

Reply:
The use case I thought was *not* changing the rate. But if we see this module mainly as an augmentation layer, users are likely to change it every batch. Then we probably shouldn't have `self.rate` at all?

Reply:
I thought this way we could do both. If someone want s a fixed rate, pass it when initializing the layer and then don't pass anything during forward, if the user wants for augmentation then every pass can be different.

Reply:
Alright, how about making things clearer by their names? Like,

```python
def __init__(self, fixed_rate):
    self.layer_rate = rate  # or self.initial_rate? 

def forward(self, x, overriding_rate=None):
    if overriding_rate is None:
        rate = self.layer_rate
    else:
        rate = overriding_rate

    return functional.time_stretch(x, rate)
```

Reply:
:+1:  :+1: